### PR TITLE
Application skeleton with io_service

### PR DESCRIPTION
**Pull request by wiatrM**
This is introduction of application skeleton with IO service threaded support.
This is work-in progress PR:

- [x] Stoppable class
- [x] Application interface
- [x] IOService
- [x] ThreadPool for job queue ~~and beast support~~ - will be moved to another PR
- [x] Some more UT

### CODE REVIEW

**Comment by wiatrM** [comment]
Plz review again

File: kazesl/core/impl/IOService.cpp [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ */
+
+#include <core/IOService.hpp>
+
+namespace kazesl
+{
+
+IOService::IOService(std::size_t threads)
+{
+    std::optional<std::string> optional;
```
Comment (wiatrM) [inline kazesl/core/impl/IOService.cpp]: This will be removed, added for optional test and forgot.

File: kazesl/core/impl/Stoppable.cpp [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+}
+
+void
+Stoppable::stop()
+{
+    // We shall be started
+    assert (started_);
+
+    onStop();
+    stopped();
```
Comment (lapek) [inline kazesl/core/impl/Stoppable.cpp]: Why are you calling stopped() when onStop() does the same?

Comment (wiatrM) [reply kazesl/core/impl/Stoppable.cpp]: onStop method is designed to be overloaded by child classes. Current onStop implementation is just a default behavior if child class dont need to overload this method. So better will be to remove stopped from onStop method, good point.

File: kazesl/core/ThreadPool.hpp [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+ * Copyright (c) 2018 Kaze Organization
+ * Authors: Michał Wiatr
+ */
+
+#ifndef KAZESL_THREADPOOL
+#define KAZESL_THREADPOOL
+
+#include <algorithm>
+#include <atomic>
+//#include <future>
```
[the remark this answers is not in the document]
Comment (wiatrM) [reply kazesl/core/ThreadPool.hpp]: removed

**Pull request merged**